cloud-nio/cloud-nio-spi/src/main/scala/cloud/nio/spi/CloudNioRetry.scala

```diff
@@ -28,7 +29,7 @@ class CloudNioRetry(config: Config) {
       case Failure(exception: Exception) if isFatal(exception) => throw exception
       case Failure(exception: Exception) if !isFatal(exception) =>
         val retriesLeft = if (isTransient(exception)) maxRetries else maxRetries map { _ - 1 }
-        if (retriesLeft.forall(_ > 0)) {
+        if (retriesLeft.forall(_ >= 0)) {
```

Comment:
fixes and off by 1 issue

Reply:
Assuming the original intent was "max retries" rather than "max tries". :)

Good catch!